templates/android/docs/java/example.md.twig

```diff
@@ -1,93 +1,63 @@
+    @NotNull
+    @Override
+    public CoroutineContext getContext() {
+        return EmptyCoroutineContext.INSTANCE;
+    }
 
-            @Override
-            public void resumeWith(@NotNull Object o) {
-                String json = "";
-                try {
-                    if (o instanceof Result.Failure) {
-                        Result.Failure failure = (Result.Failure) o;
-                        throw failure.exception;
-                    }{% if method.type != "webAuth" %} else {
-                            Response response = (Response) o;
-                            json = response.body().string();
-                        }{% endif %}
-                    
-                    }
-                } catch (Throwable th) {
-                    Log.e("ERROR", th.toString());
-                }
-            }
-        });{% endif %}
+    @Override
+    public void resumeWith(@NotNull Object o) {
+        if (o instanceof Result.Failure) {
+            Result.Failure failure = (Result.Failure) o;
+            failure.exception.printStackTrace();
+            return;
+        }
+
+        {{ utils.resultTypeJava(sdk.namespace, method) }} response = ({{ utils.resultTypeJava(sdk.namespace, method) }}) o;
+    }
+});{% endif %}
 
-{% if method.type == "webAuth" %}
-            this,
-{% endif %}
 {% for parameter in method.parameters.all %}
 {% if parameter.required %}
-            {{ parameter | paramExample }}{% if not loop.last %}, {% endif %}
+    {{ parameter | paramExample }}{% if not loop.last %},{% endif %}
 
 {% endif %}
 {% if loop.last %}
-            new Continuation<Object>() {
-                @NotNull
-                @Override
-                public CoroutineContext getContext() {
-                    return EmptyCoroutineContext.INSTANCE;
-                }
-
-                @Override
-                public void resumeWith(@NotNull Object o) {
-                    String json = "";
-                    try {
-                        if (o instanceof Result.Failure) {
-                            Result.Failure failure = (Result.Failure) o;
-                            throw failure.exception;
-                        }{% if method.type != "webAuth" %} else {
-                            Response response = (Response) o;
-                            json = response.body().string();
-                        }{% endif %}
-                    
-                    } catch (Throwable th) {
-                        Log.e("ERROR", th.toString());
-                    }
-                }
+    new Continuation<{{ utils.resultTypeJava(sdk.namespace, method) }}>() {
+        @NotNull
+        @Override
+        public CoroutineContext getContext() {
+            return EmptyCoroutineContext.INSTANCE;
+        }
+
+        @Override
+        public void resumeWith(@NotNull Object o) {
```

Comment:
Unfortunately, it can't be, due to the way Java interops with inline classes in Kotlin. In practice `o` will be typed as either a `Result.Failure` or the generic type of the continuation, which can't be represented in any way other than `Object` in Java. That's also why we have the `instanceof Result.Failure` check